File: Week_03/50/powx_n.go
package main

func myPow(x float64, n int) float64 {
	if n == 0 {
		return 1
	} else if n == 1 {
		return x
	}

	if n < 0 {
		n = -n
		x = 1/x
	}
	sub := myPow(x, n/2)
	if n%2 == 0 {
		return sub * sub
	} else {
		return sub * sub * x
	}
}

File: Week_03/50/powx_n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0:
            return 1
        elif n == 1:
            return x
        elif n < 0:
            n = -n
            x = 1 / x

        sub = self.myPow(x, n // 2)
        if n % 2 == 0:
            return sub * sub
        else:
            return sub * sub * x
